Core/src/org/sleuthkit/autopsy/casemodule/MultiUserCase.java

```diff
@@ -110,34 +121,60 @@ Date getLastAccessedDate() {
      * @return See CaseStatus enum definition.
      */
     CaseStatus getStatus() {
-        if (AutoIngestAlertFile.exists(caseDirectoryPath)) {
+        if(caseDirectoryPath.resolve("autoingest.alert").toFile().exists()) {
             return CaseStatus.ALERT;
         } else {
             return CaseStatus.OK;
         }
     }
 
     /**
+     * Gets the case metadata from a case directory path.
+     *
+     * @param caseDirectoryPath The case directory path.
+     *
+     * @return Case metadata.
+     */
+    static CaseMetadata getCaseMetadataFromCaseDirectoryPath(Path caseDirectoryPath) throws CaseMetadata.CaseMetadataException {
+        CaseMetadata caseMetadata = null;
+        
+        File directory = new File(caseDirectoryPath.toString());
+        if (directory.isDirectory()) {
+            String fileNamePrefix = directory.getName();
+            if (TimeStampUtils.endsWithTimeStamp(fileNamePrefix)) {
+                fileNamePrefix = fileNamePrefix.substring(0, fileNamePrefix.length() - TimeStampUtils.getTimeStampLength());
+            }
```

Comment:
Although I revised this code, I'm rethinking it. There's no guarantee that if the path has a time stamp, the AUT file won't. A full directory scan for an AUT file would avert this odd scenario, but it may be slower too.

Reply:
New commit addresses this.

Reply:
Have you profiled the isDirectory call, plus the regular expression pattern matching, plus the string concatenation, plus File object construction, plus the File.isFile calls to see if there is a time savings? This seems like premature optimization at the cost of greater code complexity, anmd the complexity of the code has already led to a possible NPE bug (See below). Let's just keep it simple and do the file listing. Then you could even consider inlining this code. If we are having a performance problem, then we can profile and

Reply:
I am confused. Why is this code still doing anything other than a simple file listing to find the case metadata file?